langchain/utilities/zapier.py

```diff
@@ -140,10 +162,30 @@ def run(
         """
         session = self._get_session()
         request = self._get_action_request(action_id, instructions, params)
+        print("session.prepare_request(request)",
+              session.prepare_request(request))
         response = session.send(session.prepare_request(request))
         response.raise_for_status()
         return response.json()["result"]
 
+    async def arun(self, action_id: str, instructions: str, params: Optional[Dict] = None) -> Dict:
+        """Async version of run. Executes an action with the given instructions
+        and optional parameters, returning the JSON result of the action."""
+        request_data = await self._aget_action_request(instructions, params)
+        url = f"https://nla.zapier.com/api/v1/exposed/{action_id}/execute/"
```

Comment:
Let's make a method

```
def _get_action_request_uri(self, action_id: str) -> str:
    return f"{self.zapier_nla_api_base}exposed/{action_id}/execute/"
```

And use that both in _aget_action_request and _get_action_request. We ought to make the methods as parallel as possible as well, which would mean changing the _aget_action_request  signature to return a `Request` object if possible

Reply:
Based on your suggestion below, I have added a function  
```
 def _get_action_request_uri(self, action_id: str) -> str:
    return f"{self.zapier_nla_api_base}exposed/{action_id}/execute/" ```